File: scripts/problem_template/template.py
from typing import List


class Solution:
    def solutionMethod(self, param: List[int]) -> bool:
        """
        Problem PROBLEM_NUMBER: [Problem Title]

        [Problem Description]

        Constraints:
            - [Add constraints here]
        """

        # TODO: Implement solution
        pass


if __name__ == "__main__":
    # Example usage
    solution = Solution()
    problem_number = PROBLEM_NUMBER

    def test_case(case_num: int, param, expected_result):
        """Helper function to run test cases with detailed error messages"""
        result = solution.solutionMethod(param)
        try:
            assert result == expected_result
            print(f"✅ Test case {case_num} passed")
        except AssertionError:
            print(f"❌ Problem {problem_number} - Test case {case_num} failed:")
            print(f"   Parameter: {param}")
            print(f"   Expected:  {expected_result}")
            print(f"   Got:       {result}")
            raise

    # Test case 1
    # param = [example_input]
    # expected_result = expected_output
    # test_case(1, param, expected_result)

    # Test case 2
    # param = [example_input]
    # expected_result = expected_output
    # test_case(2, param, expected_result)

    # Test case 3
    # param = [example_input]
    # expected_result = expected_output
    # test_case(3, param, expected_result)

    print(f"🎉 Problem {problem_number}: All test cases passed!")

File: scripts/problem_template/template.go
package main

import (
	"fmt"
)

// Problem PROBLEM_NUMBER: [Problem Title]
//
// [Problem Description]
//
// Constraints:
//     - [Add constraints here]
func solutionFunction(param []int) bool {
	// TODO: Implement solution
	return false
}

// Helper function to run test cases with detailed error messages
func testCase(caseNum int, problemNum int, param []int, expectedResult bool) {
	result := solutionFunction(param)
	if result == expectedResult {
		fmt.Printf("✅ Test case %d passed\n", caseNum)
	} else {
		fmt.Printf("❌ Problem %d - Test case %d failed:\n", problemNum, caseNum)
		fmt.Printf("   Parameter: %v\n", param)
		fmt.Printf("   Expected:  %v\n", expectedResult)
		fmt.Printf("   Got:       %v\n", result)
		panic("Test failed")
	}
}

func main() {
	problemNumber := PROBLEM_NUMBER

	// Test case 1
	// param := []int{example_input}
	// expectedResult := expected_output
	// testCase(1, problemNumber, param, expectedResult)

	// Test case 2
	// param = []int{example_input}
	// expectedResult = expected_output
	// testCase(2, problemNumber, param, expectedResult)

	// Test case 3
	// param = []int{example_input}
	// expectedResult = expected_output
	// testCase(3, problemNumber, param, expectedResult)

	fmt.Printf("🎉 Problem %d: All test cases passed!\n", problemNumber)
}
